llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -9635,7 +9629,6 @@ void VPDerivedIVRecipe::execute(VPTransformState &State) {
       State.Builder, CanonicalIV, getStartValue()->getLiveInIRValue(), Step,
       Kind, cast_if_present<BinaryOperator>(FPBinOp));
   DerivedIV->setName(Name);
-  assert(DerivedIV != CanonicalIV && "IV didn't need transforming?");
```

Comment:
Such redundant recipes should be cleaned up, if generated at all?

Reply:
Cleaned up in the latest version, except for the case where the vector trip count is constant null, which will only be set in prepareToExecute.

Reply:
Yep